pkg/lightning/backend/local/duplicate.go

```diff
@@ -720,7 +720,7 @@ func (m *DupeDetector) buildDupTasks() ([]dupTask, error) {
 		putToTaskFunc(ranges, nil)
 	})
 	for _, indexInfo := range m.tbl.Meta().Indices {
-		if indexInfo.State != model.StatePublic {
+		if indexInfo.State != model.StatePublic || !indexInfo.Unique {
```

Comment:
please also check if the clustered / non-clustered PK will appear in `m.tbl.Meta().Indices` and their `.Unique`